### DISCUSSION TITLE
"Expected REAL for item 1 in formatted transfer, got INTEGER" Reading lai from .nc

### DISCUSSION THREAD

**Discussion by ViniciusBMoreira** [mhm-ufz/mHM#24] [Q&A] [answered]
Hey guys

First of all, thanks for the help you always have!

Now that I've been able to run the model, I'm testing several sources for input data.

But I got this error when trying to run the model with a lai.nc file

I know it's a fortran error, not a model error. But I believe I created the lai.nc file following all the specifications

lai.nc and mhm.nml : 
[lai_mhm.zip](https://github.com/mhm-ufz/mhm/files/6697881/lai_mhm.zip)

```
  Read data ...
  Reading data ...
    Reading dem and lcover ...
      Reading dem for domain: 1 ...
      Reading lcover for domain: 1 ...
    Reading data for domain: 1 ...
      Reading slope ...
      Reading aspect ...
      Reading soil ids ...
      Reading LAI ...
At line 804 of file /mnt/c/Users/vinicius/Documents/ufmg/tcc/mhm/mhm-5.11.1/src/lib/mo_string_utils.f90
Fortran runtime error: Expected REAL for item 1 in formatted transfer, got INTEGER
(F7.2)
 ^

Error termination. Backtrace:
#0  0x7f3ef0600d3a
#1  0x7f3ef0601849
#2  0x7f3ef060252f
#3  0x7f3ef0837bc1
#4  0x7f3ef0844752
#5  0x7f3ef0847f45
#6  0x7f3ef0848573
#7  0x7f3ef0bf040c
#8  0x7f3ef09d1b72
#9  0x7f3ef09d7bbe
#10  0x7f3ef0c0c9db
#11  0x7f3ef0c0e8a6
#12  0x7f3ef02970b2
#13  0x7f3ef0982c7d
#14  0xffffffffffffffff
```

**Comment by LuvMangorange** [comment, marked as answer]
Hi @BryanSouzaM 
The error maybe come from the format of the time dimension in your lai.nc data.
The time format should be integer, but yours is not.
![image](https://user-images.githubusercontent.com/34867664/123084140-6235d000-d453-11eb-9a55-9103d73b7485.png)
![image](https://user-images.githubusercontent.com/34867664/123084194-72e64600-d453-11eb-87b3-c41930ebdfae.png)

By the way, I have a question about the process of LAI (from Modis). How to deal with the fillvalue ’255‘ and the missing value?
Thanks for help!

**Reply by rohinikumar82** [reply]
Hi @LuvMangorange,

thanks for helping our Bryan - good to see this community effort building up. For your LAI related question - I would say you can do it in two ways - either fill with the nearest (valid) values or give a small number (like 0.0001 or so) - the latter you can choose for the case where the landcover is a not dominant vegetation type (but for example, desert or water-bodies).. A very small number would let the mHM to run smoothly without getting into issue of no-data value. 

Hope this helps...

Best,
Rohini

**Reply by LuvMangorange** [reply]
Hi @rohinikumar82 
Thank you for the hlep. What I have done is the latter you suggested. 
By the way, I also have a doubt about my geology_class data. In other word, I would like to get some suggestions or comments from you on my method of making the gology_class file. 
The gology_class.asc file is related to k2(the mean recession constant), and k2 has a strong correlation with hydrologic soil group(Hoffmann, J. G. ,  Mangarillo, J. T. , &  Brandes, D. . (2014). Correlation of Multiple Baseflow Recession Constants with Watershed Hydrogeomorphic Parameters. World Water & Environmental Resources Congress.). Therefore, I make the geology_class.asc file based on the H.S.G file which generated by soil data. As a result, my sample does not contain any rock-related data. So I have some doubts about this approach. 
 Looking forward to your reply!

**Reply by rohinikumar82** [reply]
Dear @LuvMangorange,
If you have a good reasoning for K2 (that directly effect baseflow) to be related with soil group types - then you can use those groupings for the gology_class.asc (and also for the LUT file). I would just suggest that not to make too many classes - take something between 5-10 classes - as these will increase your calibration parameters (and thereby parameter interactions and other issues).
Best,
Rohini
